includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WP_List_Table' ) ) {
+	require_once( ABSPATH . 'wp-admin/includes/class-wp-list-table.php' );
+}
+
+/**
+ * WC_Report_Downloads.
+ *
+ * @author      WooThemes
+ * @category    Admin
+ * @package     WooCommerce/Admin/Reports
+ * @version     3.3.0
+ */
+class WC_Report_Downloads extends WP_List_Table {
+
+	/**
+	 * Max items.
+	 *
+	 * @var int
+	 */
+	protected $max_items;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		parent::__construct( array(
+			'singular'  => 'download',
+			'plural'    => 'downloads',
+			'ajax'      => false,
+		) );
+	}
+
+	/**
+	 * Don't need this.
+	 *
+	 * @param string $position
+	 */
+	public function display_tablenav( $position ) {
+
+		if ( 'top' !== $position ) {
+			parent::display_tablenav( $position );
+		}
+	}
+
+	/**
+	 * Output the report.
+	 */
+	public function output_report() {
+
+		$this->prepare_items();
+
+		// Subtitle for permission if set
+		if ( isset( $_GET['permission_id'] ) && is_numeric( $_GET['permission_id'] ) && $_GET['permission_id'] > 0 ) {
+			// Load the permission, order, etc. so we can render more information
+			$permission	= null;
+			$product	= null;
+			try {
+				$permission	= new WC_Customer_Download( $_GET['permission_id'] );
+				$product	= wc_get_product( $permission->product_id );
+			} catch ( Exception $e ) {
+				echo '<p>Permission #' . esc_html( $_GET['permission_id'] ) . ' not found.</p>';
+			}
+
+			if ( ! empty( $permission ) && ! empty( $product ) ) {
+				echo '<h2>';
+
+				// File information
+				if ( $file = $product->get_file( $permission->get_download_id() ) ) {
+					echo esc_html( $file->get_name() ) . ' for ';
+				}
+
+				// Product information
+				$product_description = '';
+				if ( $sku = $product->get_sku() ) {
+					$product_description .= $sku . ' - ';
+				}
+				$product_description .= $product->get_name();
+				echo edit_post_link( $product_description, '', '', $product->get_id() );
+
+				// Order information
+				echo ' in Order ';
+				echo edit_post_link( '#' . $permission->order_id, '', '', $permission->order_id );
```

Comment:
Actually I think the edit_post_link function emits the full `A` tag with the url included and passed through `esc_url` already, so should be good on this one (let me know if I'm misunderstanding though).  I'll remove the `echo` though since it's already included.